pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -157,23 +157,25 @@ func (gb *GraphBuilder) controllerFor(resource schema.GroupVersionResource, kind
 	return monitor, nil
 }
 
-func (gb *GraphBuilder) monitorsForResources(resources map[schema.GroupVersionResource]struct{}) error {
+func (gb *GraphBuilder) monitorsForResources(resources map[schema.GroupVersionResource]struct{}) {
 	for resource := range resources {
 		if _, ok := ignoredResources[resource]; ok {
 			glog.V(5).Infof("ignore resource %#v", resource)
 			continue
 		}
 		kind, err := gb.restMapper.KindFor(resource)
 		if err != nil {
-			return err
+			nonCoreMsg := fmt.Sprintf(nonCoreMessage, resource)
+			utilruntime.HandleError(fmt.Errorf("%v. %s", err, nonCoreMsg))
+			continue
 		}
 		monitor, err := gb.controllerFor(resource, kind)
 		if err != nil {
-			return err
+			utilruntime.HandleError(err)
```

Comment:
Only possible error is the failure to get a dynamic client for a kind. I think should fail the initialization if that happens. Will update.